R/decisiontreeutils.R

```diff
@@ -0,0 +1,363 @@
+  library(AlteryxRviz)
+  library(htmltools)
+  if (is_XDF){
+    k1 = tags$div(tags$h4(
+      "Interactive Visualizations are not supported for Revolution Enterprise"
+    ))
+    # renderInComposer(k1, nOutput = 5)
+  } else {
+    if (!(packageVersion('AlteryxRviz') >= "0.2.5")){
+      k1 = tags$div(
+        tags$h4("You need AlteryxRviz >= 0.2.5")
+      )
+    } else {
+      #the_model = rpart(Species ~ ., data = iris)
+      tooltipParams = list(
+        width = '250px',
+        top = '130px',
+        left = '100px'
+      )
+      dt = renderTree(the_model, tooltipParams = tooltipParams)
+      vimp = varImpPlot(the_model, height = 300)
+
+      cmat = if (!is.null(the_model$frame$yval2)){
+        iConfusionMatrix(getConfMatrix(the_model), height = 300)
+      }  else {
+        tags$div(h1('Confusion Matrix Not Valid'), height = 300)
+      }
+
+      k1 = dtDashboard(dt, vimp, cmat)
+    }
+    # renderInComposer(k1, nOutput = 5)
+  }
+  k1
+}
+
+#' get results in form of output list
+#'
+#' @param config list of config options
+#' @param the_model model object
+#' @param is_XDF boolean of whether model is XDF
+getOutputsDT <- function(config, the_model, is_XDF) {
+  # Assemble list to return needed elements to output
+  results <- list()
+
+  results$output1 <- getDTPipes(config, the_model, is_XDF)
+  results$output3 <- modelOut(config$model.name, the_model)
+
+  graph_results <- getDTgraphCalls(config, the_model, is_XDF)
+  results$output2 <- graph_results$tree
+  results$output4 <- graph_results$prune
+
+  results$output5 <- getDTViz(the_model, is_XDF)
+
+  write.Alteryx(results$output1, nOutput = 1)
+  write.Alteryx(results$output3, nOutput = 3)
+
+  renderInComposer(results$output5, nOutput = 5)
+
+  results
+}
+
+#' output results to Alteryx
+#'
+#' @param results list of results to output
+#' @param config list of config options
+outputDTResultsAlteryx <- function(results, config) {
+  write.Alteryx(results$output1, 1)
+  write.Alteryx(results$output2, 2)
+  renderInComposer(results$output5, 5)
+
+  whr <- graphWHR(inches = config$tree.inches, in.w = config$tree.in.w, in.h = config$tree.in.h, cm.w = config$tree.cm.w, cm.h = config$tree.cm.h, resolution = config$tree.graph.resolution, print.high = TRUE)
+  AlteryxGraph(2, width = whr[1], height = whr[2], res = whr[3], pointsize = config$tree.pointsize)
+  par(mar = c(5, 4, 6, 2) + 0.1)
+  do.call(match.fun(results$output2$f), results$output2$args)
+  invisible(dev.off())
+
+  whr <- graphWHR(inches = config$prune.inches, in.w = config$prune.in.w, in.h = config$prune.in.h, cm.w = config$prune.cm.w, cm.h = config$prune.cm.h, resolution = config$prune.graph.resolution, print.high = FALSE)
+  AlteryxGraph(4, width = whr[1], height = whr[2], res = whr[3], pointsize = config$prune.pointsize)
+  par(mar = c(5, 4, 6, 2) + 0.1)
+  do.call(match.fun(results$output4$f), results$output4$args)
+  title(main="Pruning Plot", line=5)
+  invisible(dev.off())
+}
```

Comment:
Good. That function belongs to the macro, since I believe the package should be free of Alteryx I/O. This allows someone to update outputs without having to hack at the package.